lib/stdlib/src/base64.erl

```diff
@@ -140,367 +158,379 @@ mime_decode(List) when is_list(List) ->
+    mime_decode_list(ModeOffset, Cs, A, B1, B2);
+mime_decode_list(ModeOffset, [C3 | Cs], A, B1, B2) ->
+    case b64d(C3, ModeOffset) of
         B3 when is_integer(B3) ->
-            mime_decode_list(Cs, A, B1, B2, B3);
+            mime_decode_list(ModeOffset, Cs, A, B1, B2, B3);
         eq=B3 ->
-            mime_decode_list_after_eq(Cs, A, B1, B2, B3);
-        _ -> mime_decode_list(Cs, A, B1, B2)
+            mime_decode_list_after_eq(ModeOffset, Cs, A, B1, B2, B3);
+        _ -> mime_decode_list(ModeOffset, Cs, A, B1, B2)
     end.
 
-mime_decode_list([0 | Cs], A, B1, B2, B3) ->
-    mime_decode_list(Cs, A, B1, B2, B3);
-mime_decode_list([C4 | Cs], A, B1, B2, B3) ->
-    case b64d(C4) of
+mime_decode_list(ModeOffset, [0 | Cs], A, B1, B2, B3) ->
+    mime_decode_list(ModeOffset, Cs, A, B1, B2, B3);
+mime_decode_list(ModeOffset, [C4 | Cs], A, B1, B2, B3) ->
+    case b64d(C4, ModeOffset) of
         B4 when is_integer(B4) ->
-            mime_decode_list(Cs, <<A/bits,B1:6,B2:6,B3:6,B4:6>>);
+            mime_decode_list(ModeOffset, Cs, <<A/bits,B1:6,B2:6,B3:6,B4:6>>);
         eq ->
-            mime_decode_list_after_eq(Cs, A, B1, B2, B3);
-        _ -> mime_decode_list(Cs, A, B1, B2, B3)
+            mime_decode_list_after_eq(ModeOffset, Cs, A, B1, B2, B3);
+        _ -> mime_decode_list(ModeOffset, Cs, A, B1, B2, B3)
     end.
 
-mime_decode_list_after_eq([0 | Cs], A, B1, B2, B3) ->
-    mime_decode_list_after_eq(Cs, A, B1, B2, B3);
-mime_decode_list_after_eq([C | Cs], A, B1, B2, B3) ->
-    case b64d(C) of
+mime_decode_list_after_eq(ModeOffset, [0 | Cs], A, B1, B2, B3) ->
+    mime_decode_list_after_eq(ModeOffset, Cs, A, B1, B2, B3);
+mime_decode_list_after_eq(ModeOffset, [C | Cs], A, B1, B2, B3) ->
+    case b64d(C, ModeOffset) of
         B when is_integer(B) ->
             %% More valid data, skip the eq as invalid
             case B3 of
-                eq -> mime_decode_list(Cs, A, B1, B2, B);
-                _ -> mime_decode_list(Cs, <<A/bits,B1:6,B2:6,B3:6,B:6>>)
+                eq -> mime_decode_list(ModeOffset, Cs, A, B1, B2, B);
+                _ -> mime_decode_list(ModeOffset, Cs, <<A/bits,B1:6,B2:6,B3:6,B:6>>)
             end;
-        _ -> mime_decode_list_after_eq(Cs, A, B1, B2, B3)
+        _ -> mime_decode_list_after_eq(ModeOffset, Cs, A, B1, B2, B3)
     end;
-mime_decode_list_after_eq([], A, B1, B2, eq) ->
+mime_decode_list_after_eq(_ModeOffset, [], A, B1, B2, eq) ->
     <<A/bits,B1:6,(B2 bsr 4):2>>;
-mime_decode_list_after_eq([], A, B1, B2, B3) ->
+mime_decode_list_after_eq(_ModeOffset, [], A, B1, B2, B3) ->
     <<A/bits,B1:6,B2:6,(B3 bsr 2):4>>.
 
-mime_decode_binary(<<0:8, Cs/bits>>, A) ->
-    mime_decode_binary(Cs, A);
-mime_decode_binary(<<C1:8, Cs/bits>>, A) ->
-    case b64d(C1) of
-        B1 when is_integer(B1) -> mime_decode_binary(Cs, A, B1);
-        _ -> mime_decode_binary(Cs, A)  % eq is padding
+mime_decode_binary(ModeOffset, <<0:8, Cs/bits>>, A) ->
+    mime_decode_binary(ModeOffset, Cs, A);
```

Comment:
If you change the decode offsets to be 1, 256, and 512 (with the corresponding adjustments to the tuple), you can omit this clause and all similar clauses.

```suggestion
```

Reply:
Done (but it is 257 and 513 ;))